client/components/bulk-select/index.jsx

```diff
@@ -21,33 +19,30 @@ export default React.createClass( {
 	},
 
 	getStateIcon() {
-		if ( this.hasSomeElementsSelected() ) {
+		if ( ! this.hasAllElementsSelected() && this.hasSomeElementsSelected() ) {
 			return <Gridicon className="bulk-select__some-checked-icon" icon="minus-small" size={ 18 }/>;
 		}
 	},
 
 	hasAllElementsSelected() {
-		return this.props.selectedElements && this.props.selectedElements === this.props.totalElements;
+		return this.props.selectedElements === this.props.totalElements;
 	},
 
 	hasSomeElementsSelected() {
-		return this.props.selectedElements && this.props.selectedElements < this.props.totalElements;
+		return this.props.selectedElements > 0;
 	},
 
 	handleToggleAll() {
-		const newCheckedState = ! ( this.hasSomeElementsSelected() || this.hasAllElementsSelected() );
-		this.props.onToggle( newCheckedState );
+		this.props.onToggle( this.hasSomeElementsSelected() );
 	},
 
 	render() {
 		const isChecked = this.hasAllElementsSelected();
-		const inputClasses = classNames( 'bulk-select__box', {
-			'is-checked': isChecked // we need to add this css class to be able to test if the input if checked, since enzyme still doesn't support :checked pseudoselector
```

Comment:
enzyme does allow querying by attribute so `{checked: true}` works in the tests.